zerver/actions/message_send.py

```diff
@@ -210,11 +214,25 @@ def get_recipient_info(
         # of this function for different message types.
         assert stream_topic is not None
 
+        possibly_topic_wildcard_mentioned_user_ids: AbstractSet[int] = set()
+        if possible_topic_wildcard_mention:
+            # Anyone who sent a message to the topic or reacted to a message on the topic
+            # is considered a possible user to be notified for topic mention.
+            messages = messages_for_topic(recipient.id, stream_topic.topic_name)
+            possibly_topic_wildcard_mentioned_user_ids = (
+                {message.sender_id for message in messages}
+            ) | set(
+                Reaction.objects.filter(message__in=messages).values_list(
+                    "user_profile_id", flat=True
+                )
+            )
```

Comment:
This will make 2 database queries, one to fetch the sender IDs, and one to fetch the reactor IDs.
Can we avoid that and do a single query on the UserProfile table, maybe using Django's `Q` objects?
Ref: https://docs.djangoproject.com/en/4.2/topics/db/queries/#complex-lookups-with-q-objects